iml-manager-cli/src/api.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlManagerCliError;
+use console::Term;
+use structopt::{clap::arg_enum, StructOpt};
+
+arg_enum! {
+    #[derive(PartialEq, Debug)]
+    pub enum ApiType {
+        Delete,
+        Get,
+        Post,
+        Put,
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct ApiCommand {
+    #[structopt(possible_values = &ApiType::variants(), case_insensitive = true)]
+    call: ApiType,
+
+    path: String,
+
+    /// Object to PUT
```

Comment:
nit: could be any method actually, at least POST and PUT. Better be called `body`.